docker/django-server.override.yml

```diff
@@ -5,6 +5,8 @@ services:
     build:
       context: ..
       dockerfile: docker/Dockerfile
+      args:
+        DEVELOPMENT: true
```

Comment:
honestly I do not like to have this set as default here. I mean, in this way it is not possible to customize it and effectively run the code in the `repo_downloader.sh` file. If we need to do that we would find ourselves in the same problem as before where we change manually a file with the risk to commit it unwillingly.

The only way to solve this issue would be to add the chance to choose whether to run that bash script or not via one of the file that are NOT commited (`env_file_app`, `.env`)

Reply:
Even I thought that. However, in the same override file, we are also setting the ```DEBUG``` and ```DJANGO_TEST_SERVER``` variables too. Should we look towards migrating them too?

Reply:
no, I mean that is different. If you are in development mode, you always need those 2 variables set.

On the contrary, for the `repo_downloader.sh`, you could need to have it running if you are working on it but most of the time you won't.

Reply:
I have created a new file in the root called ```.env.test.template``` and added ```.env.test``` to gitignore. Now the developer can freely modify ```.env.test``` without having the risk of accidentally committing it to the repo. Does that look good?

Reply:
@mlodic I've also verified it to be working and will shortly post screenshots of the same

Reply:
there is no need to create new environement files and use the `dotenv` library. As I mentioned earlier, you could use the already existing ones...please check inside the `docker` folder

Reply:
@mlodic the ```docker/env_file_*``` files are used to set environment variables. However, here, we need to set a build argument (that determines if the script will be executed or not). The only plausible way to set this argument is from ```start.py```. Hence, I had to create a new .env file to be sourced by the same.

Reply:
thanks for the explanation. I got you. that0s ok so.

Just 2 other things:
* is it possible to add some documentation about this? (maybe in the "hint" section [here](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance)?)
* other concern is that we are adding another new file to the main directory. We risk to clutter it too much so when it is not 100% needed, we should try to move it somewhere else. Even if it is not docker-related, I think it would make sense to move that new env file in the `docker` folder with the others and change its name in something more intuitive (otherwise we risk to confuse it with the .env file of docker). Maybe `env_start_test` and `env_start_test_template`?

Reply:
@mlodic I was wary of using underscore in the name as that could confuse it with docker-compose's ```env_file```s. How is ```.env.start.test```. ```start``` because it is sourced by ```start.py``` and ```test``` as it is meant for test mode

Reply:
Should the documentation be in this issue itself or should I create a separate issue for the same?

Reply:
related documentation is always part of the issue itself otherwise we forget to do that :P

Reply:
@mlodic resolved all concerns